caluma/caluma_form/tests/test_document.py

```diff
@@ -1444,3 +1444,77 @@ def test_document_modified_content_properties(
     assert node["modifiedContentAt"] == sub_a.modified_at.isoformat()
     assert node["modifiedContentByUser"] == sub_a.modified_by_user
     assert node["modifiedContentByGroup"] == sub_a.modified_by_group
+
+
+@pytest.mark.parametrize(
+    "question__type,answer__value",
+    [
+        (Question.TYPE_CHOICE, "somevalue"),
+        (Question.TYPE_MULTIPLE_CHOICE, ["somevalue", "anothervalue"]),
+        (Question.TYPE_DYNAMIC_CHOICE, "5.5"),
+        (Question.TYPE_DYNAMIC_MULTIPLE_CHOICE, ["5.5"]),
+        (Question.TYPE_TEXT, "somevalue"),
+    ],
+)
+def test_selected_options(
+    db,
+    snapshot,
+    document,
+    answer,
+    question_option_factory,
+    dynamic_option_factory,
+    schema_executor,
+    question,
+):
+    query = """
+        query AllDocumentsQuery($search: String) {
+          allDocuments(search: $search) {
+            edges {
+              node {
+                answers {
+                  edges {
+                    node {
+                      ... on StringAnswer {
+                        string_value: value
+                        selectedOption {
+                          slug
+                          label
+                        }
+                      }
+                      ... on ListAnswer {
+                        list_value: value
+                        selectedOptions {
+                          edges {
+                            node {
+                              slug
+                              label
+                            }
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+    """
+
+    search = isinstance(answer.value, list) and " ".join(answer.value) or answer.value
+
+    if question.type == Question.TYPE_CHOICE:
+        question_option_factory(option__slug=answer.value, question=question)
+    elif question.type == Question.TYPE_DYNAMIC_CHOICE:
+        dynamic_option_factory(slug=answer.value, question=question, document=document)
+
+    elif question.type == Question.TYPE_MULTIPLE_CHOICE:
+        for slug in answer.value:
+            question_option_factory(option__slug=slug, question=question)
+    elif question.type == Question.TYPE_DYNAMIC_MULTIPLE_CHOICE:
+        for slug in answer.value:
+            dynamic_option_factory(slug=slug, question=question, document=document)
+
+    result = schema_executor(query, variable_values={"search": search})
+    assert not result.errors
+    snapshot.assert_match(result.data)
```

Comment:
I'd like to see some explicit checks here as well. Sometimes we don't look at the snapshots too closely and may miss something

Reply:
Done.